### PR TITLE
Visitor EV3

### PR DESCRIPTION
**Pull request by aykborstelmann**
This is a PR for the visitor changes in the EV3 Plugin.

Things needing to be discussed (@rbudde)
- [x] Do we need to rethink the Collector structure. Meaning should the interface `SC` be refactored to a enum and should all properties in it be documentated ? Problem is e.g. in visitPlayFileAction should I register a UsedActor with `SC.SOUND` or `SC.MUSIC`?
- [x] When adding a internal UsedActor we use `- EMPTY_PORT -` 
- [x] Added Collector Tests like the `EV3ProgramUsedHardwareCheckTest` in `ReuseIntegrationAsUnitTest`
- [x] In DifferentialMotorValidatorAndCollectorVisitor#visitTurnAction no warning for Zero speed if there is no duration
- [x] Call in `DifferentialMotorValidatorAndCollectorVisitor`  `addLeftAndRightMotorToUsedActors()` in every method ?

### CODE REVIEW

**Comment by aykborstelmann** [comment]
I need to review this by myself, but I think this is finished.

File: RobotEV3/src/main/java/de/fhg/iais/roberta/visitor/validate/Ev3SimValidatorAndCollectorVisitor.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    public Ev3SimValidatorAndCollectorVisitor(
+        ConfigurationAst robotConfiguration,
+        ClassToInstanceMap<IProjectBean.IBuilder<?>> beanBuilders) {
+        super(robotConfiguration, beanBuilders);
+    }
+
+    @Override
+    public Void visitBluetoothReceiveAction(BluetoothReceiveAction<Void> bluetoothReceiveAction) {
+        addWarningToPhrase(bluetoothReceiveAction, "SIM_BLOCK_NOT_SUPPORTED");
+        return super.visitBluetoothReceiveAction(bluetoothReceiveAction);
```
Comment (aykborstelmann) [inline RobotEV3/src/main/java/de/fhg/iais/roberta/visitor/validate/Ev3SimValidatorAndCollectorVisitor.java:22]: I'm not sure why the call to super is necessary here. All other not supported blocks don't do this.

**Review by aykborstelmann — COMMENTED** [review]
I'm done now with the Ev3 changes, it would be great if you could look over this again, possibly together ? @rbudde

File: RobotEV3/src/main/java/de/fhg/iais/roberta/visitor/validate/Ev3SimValidatorAndCollectorVisitor.java (hunk 1 above)
Comment (rbudde) [reply RobotEV3/src/main/java/de/fhg/iais/roberta/visitor/validate/Ev3SimValidatorAndCollectorVisitor.java:22]: super is needed if there is a sub structure

**Pull request merged**